cli/config/configuration-acrn.toml.in

```diff
@@ -32,13 +32,6 @@ kernel_params = "@KERNELPARAMS@"
 # If you want that acrn uses the default firmware leave this option empty
 firmware = "@FIRMWAREPATH@"
 
-# Default number of vCPUs per SB/VM:
-# unspecified or 0                --> will be set to @DEFVCPUS@
-# < 0                             --> will be set to the actual number of physical cores
-# > 0 <= number of physical cores --> will be set to the specified number
-# > number of physical cores      --> will be set to the actual number of physical cores
-default_vcpus = 1
```

Comment:
So, does this mean that a ACRN VM can be launched with only 1 vcpu always? 
If so, we need to document this.
You mentioned in the commit message `Number of guest vcpus will be defined in the hypervisor scenario
configuration file instead.`, but looks like with this removal, it is no longer possible, unless I am missing something.

Reply:
Hi @amshinde , yes by default ACRN can be launched with just 1vcpu per VM. So instead of updating the configration.toml file, one needs to update the vcpu option in https://github.com/projectacrn/acrn-hypervisor/blob/master/hypervisor/scenarios/logical_partition/vm_configurations.c#L20 and rebuild ACRN. This change was done for https://github.com/projectacrn/acrn-hypervisor/issues/3663.

Reply:
Rebuild ACRN and the runtime potentially according to this PR?

Could you provide more information on this. https://github.com/kata-containers/runtime/issues/2136 mentions that users will need to set the vpu number in "the hypervisor scenario configuration file", but that seems to be incorrect as I don't personally consider a C source file a configuration file.

A couple of questions:

- What are the advantages of making this change?
- How can users determine the hard-coded value baked into their acrn binary?

Reply:
/cc @egernst, @gnawux.

Reply:
Hi @jodh-intel ACRN is not a typical cloud/server hypervisor but it is mainly targeted for IoT purposes. Given that, the assumption is that for a given workload there is an ACRN configuration that remains static and doesn't change.
1. What are the advantages of making this change?
These are some of the reasons for the change,
a. Keep the logic simple and clean (ACRN needs FuSa certification, so minimize the code)
b. By removing the dynamic configuration, we eliminate vcpu migration complexities.
c. ACRN also supports other modes such as partition mode where there is no SOS. So to have a common usage model.

2. How can users determine the hard-coded value baked into their ACRN binary?
The user is the one who is building ACRN with a particular configuration and should have this information.(default is 1 vcpu per VM)

Reply:
Thanks for the info @vijaydhanraj. If I'm understanding correctly then, the hypervisor hard-codes the value of `1` but this can be overriden by setting `maxcpus` for the guest kernel. And that's what this PR is doing: `DEFMAXVCPUS_ACRN` / `default_maxvcpus` is the value that will *override* the hard-coded value of `1` in the hypervisor right?

If that sounds right, I do think it would help a little if you reworded the commit message to explain this more clearly (including either removing mention of "hypervisor scenario configuration file" or explaining that that is in fact one of the source files for the hypervisor itself - not a conventional configuration file).

Reply:
There are 2 variables we are talking about,
1. -c, --ncpus <cpus> - Set number of vCPUs for guest OS (This is an ACRN hypervisor parameter)
2. maxcpus - Maximum number of processors that an SMP kernel will bring up during bootup (this is a kernel command line parameter)

I have removed #1, and this is now defined in the hypervisor scenario configuration file. instead. Regarding 2, I have set it to 1 so that by default the guest OS will launch with 1 vcpu per VM.

So to clarify, the hypervisor hardcodes #1 based on the scenario. (Read more about scenarios here: refer: https://projectacrn.github.io/latest/introduction/#best-known-configurations, https://github.com/projectacrn/acrn-hypervisor/blob/master/hypervisor/scenarios/sdc/vm_configurations.c#L67). Say for example if the scenario is SDC, the number of vCPUs per VM is set to 2. If the user wants to limit the number of vCPUs to the guest to 1, the user can override using the DEFMAXVCPUS_ACRN / default_maxvcpus.

Also, I have reworded the commit message. Let me know if that looks fine to you.